File: SkillSync.WebServer/SkillSync.Domain/Models/ChatMessage.cs
using SkillSync.Domain.Enums;

namespace SkillSync.Domain.Models
{
    public class ChatMessage
    {
        public Guid Id { get; set; }

        public string Content { get; set; }

        public string SenderId { get; set; }
        public User Sender { get; set; }

        public DateTime SentAt { get; set; }

        public ChatMessageType Type { get; set; }
    }
}

File: SkillSync.Client/src/app/shared/entities/models/chat-message.ts
import { ChatMessageType } from "../../enums/chat-message-type";

export interface ChatMessage {
    id: string;
    content: string | File;
    sender: string;
    wasMessageSentSuccessfully?: boolean;
    sentAt: string;
    type: ChatMessageType;
    wasMediaReceivedFromStorage?: boolean;
    shouldDisplayDocument?: boolean;
}